### ISSUE TITLE
Proposal: new rule for disable importing specific items from specific module

### ISSUE THREAD

**Issue by meowtec** [import-js/eslint-plugin-import#2218]
Many packages support both "tree shacking friendly" exports style and "legacy" exports style, like this:

```js
import { add } from 'lodash'
add(1, 2)
// 2KB
```
vs

```js
import _ from 'lodash-es'
_.add(1, 2)
// 88KB
```

Need a rule to disable the second one.

## Example

Options:

```json
{
    "import/no-specific-imports": [
        "error",
        {
            "modules": [
                {
                    "name": "lodash-es",
                    "imports": [
                        "default",
                        "foo"
                    ]
                }
            ]
        }
    ]
}
```

valid:

```
import { bar } from 'lodash-es'
import type { foo } from 'lodash-es'
```

invalid:

```
import _ from 'lodash-es'
import { foo } from 'lodash-es'
import { foo, bar } from 'lodash-es'
import * as _ from 'lodash-es'
```

Draft implement:
https://github.com/meowtec/eslint-plugin-import/tree/feat/no-specific-imports

**Comment by ljharb** [comment]
Neither of those are actually tree-shaking friendly, because they both necessitate tree-shaking. The best choice is to deep import exactly what you need, and never touch the default bag-o-things at all.

Separately, it seems like lodash-es shouldn’t be providing both named and default exports - doing that is definitely not something many packages do and is absolutely an antipattern. However, Babel import interop does allow doing both, this is true.
